src/components/switch/switch.tsx

```diff
@@ -124,58 +132,65 @@ export class Switch {
     }
 
     public render() {
-        return (
-            <Host>
-                <button
-                    id={this.fieldId}
-                    class={{
-                        'mdc-switch': true,
-                        'lime-switch--readonly': this.readonly,
-                        'mdc-switch--unselected': !this.value,
-                        'mdc-switch--selected': this.value,
-                    }}
-                    type="button"
-                    role="switch"
-                    aria-checked={this.value}
-                    disabled={this.disabled || this.readonly}
-                    onClick={this.handleClick}
+        if (this.readonly) {
+            return [
+                <limel-readonly-boolean
```

Comment:
by the way @jgroth, I don't fully buy the argument that the consumer who has control over the code can also create a separate design for the readonly mode. Of course your argument is 100% valid, but with the same argument we could say that the consumer can just use a native HTML checkbox, or input field, popover, or many of the things that this component library offers. 

One of the strong points of having a UI component library is that it helps keeping the UI consistent and clean, and making it easy for a developer to make a decent UI, without taking too many layout and ui decisions about minute graphical details. 

If everyone was forced to create their own stuff for everything, then the products using this library would become messy and inconsistent. 

Also, if **A:** the `readonly` is a feature of our boolean field, and **B:** the boolean fields create this classic usability issue and lead to confusions, then it make a lot of sense that the component itself offers a solution that prevent the usability issues. We can't expect the consumer to come up with solutions. If they could, we didn't have to make this PR neither.

Reply:
> by the way @jgroth, I don't fully buy the argument that the consumer who has control over the code can also create a separate design for the readonly mode. 

They don't have to create a separate design for it. There's already a new component added in this PR that could be used (if it's made public). That also opens the possibility to use it in more scenarios than for only boolean fields. Perhaps you want this design as well if you have a readonly option field?

> One of the strong points of having a UI component library is that it helps keeping the UI consistent and clean, and making it easy for a developer to make a decent UI, without taking too many layout and ui decisions about minute graphical details.

Yes, I agree, but that does not mean that every component needs to be handle every kind of scenario. Components should ideally be small and have a single purpose, and if you need additional functionality you should create compositions of different components. This is the same problem as with the `help` property that you described above. If we need to add these kinds of properties to all the components we are bloating them and not keeping them simple.

> If everyone was forced to create their own stuff for everything, then the products using this library would become messy and inconsistent.

I'm not sure what you mean by this, but if the new component is public no one has to create their own component. We can still add functionality to lime-admin to let this be configurable without consumers creating their own components.

Reply:
> They don't have to create a separate design for it. There's already a new component added in this PR that could be used (if it's made public). That also opens the possibility to use it in more scenarios than for only boolean fields. Perhaps you want this design as well if you have a readonly option field?

I have no objection if this component is public. But I don't think its usage applies to other components in Lime Elements, than our own boolean fields. The issue it is addressing is about the boolean field.

>  that does not mean that every component needs to be handle every kind of scenario. Components should ideally be small and have a single purpose, and if you need additional functionality you should create compositions of different components. 

I _fully_ agree with your sentiments, but I don't think this is the applicable place for it. In our case we offer a component that has a feature which in turn comes an _inherent_ problem. We cannot be saying to the consumer: 

"_Hey, that we are aware of this problem, and just wanted to let you know that this problem exists. You should make sure no to have this problem when you use our component. However, if you wanna fix it, you should go use another component._"

This would be both counter intuitive and wrong. It is like buying a smartphone which has a certain feature; let say a **Turbo Mode**. But when the user turns on the Turbo Mode, the phone overheats and turns off. The manufacturer says, we know that this problem exist. But we decided not to add a cooling mechanism inbuilt for it, to keep the device more simple. If you want to use the Turbo Mode, go buy a cooler shell for your smartphone. By the way, we also produce and sell cooler shells.

We know the the readonly mode of our boolean components comes with an inherent problem. So the component itself should have means of mitigating the problem. Therefore
>  that does not mean that every component needs to be handle every kind of scenario.

is not a valid argument here. Switch and Checkbox are already handling the scenario of being readonly by changing their visual design and interactivity. But they are handling it in an insufficient way, forcing the users to have problematic experiences, or forcing the consumer to do loads of extra work to fix it.

> We can still add functionality to lime-admin to let this be configurable without consumers creating their own components.
> create a component in lime-crm-building-blocks that combines the two

This is not only a Lime Admin problem or a CRM problem. This comes with boolean fields, specially when they're set to readonly. I don't know how you want to fix it or implement this in Lime Admin. I can't do it myself anyways. You guys should find a way for it. But as I tried to explain above, I believe we need to address the issue in the component, not outside of it.

Reply:
Ok. Lots of discussion here. Sorry if I missed something by not reading all the comments. But I have a suggestion.

Is it enough to handle readonly for checkboxes in limel-form? That is, if the schema type is boolean:
- render limel-checkbox if not readonly schema
- render limel-readonly-bool if readonly in schema

Currently in Lime CRM, we're not really specifying a custom component to use for boolean fields anyway.

I guess we don't really need a component that dynamically can toggle its readonly state?

Reply:
@vicgeralds that would certainly help Lime CRM, but it wouldn't fix the inherent problem of boolean fields, when they become readonly. That'd be we just add documentations to Switch and Checkbox, and ask consumers to not just mindlessly render the boolean field in readonly mode, and instead use `limel-readonly-boolean` (which means making it public, what Johan is saying) or another proper visualization.

I don't like it too much, but I also don't see this as a show stopper. We can do so too.

Anyhow, I don't know how to fix the issue in the limel-form anyways as you describe. You guys need to do it or help me do it.